strapi/strapi
# Migrate database to ts

### What does it do?

- Migrate the database package to typescript keeping the same "almost everything is any" public interface
- Will need to improve the types more and more as they should serve as a basis for a lot of the types above them (entityService/documentService in particular)  

### How to test it?

- Overall testing of the app

Areas of QA:

- Relations reordering (both sides)
- Data transfer use of the db might be impacted
- Custom/plugin code using strapi.query or strapi.db.xxx should not get type errors


### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
